higher_health/forms.py

```diff
@@ -347,13 +339,37 @@ def medical_fields(self):
 
 
 class HealthCheckLogin(forms.Form):
-    phone = forms.CharField(
-        widget=TextInput(attrs={"placeholder": "Phone number"}),
+    msisdn = forms.CharField(
+        label="Enter your mobile number",
+        widget=TextInput(attrs={"placeholder": "Mobile number"}),
         required=True,
-        max_length=100,
+        validators=[za_phone_number],
     )
-    fullname = forms.CharField(
-        widget=TextInput(attrs={"placeholder": "Last name"}),
+
+    def clean_msisdn(self):
+        number = phonenumbers.parse(self.cleaned_data["msisdn"], "ZA")
+        return phonenumbers.format_number(number, phonenumbers.PhoneNumberFormat.E164)
+
+
+class HealthCheckOTP(forms.Form):
+    otp = forms.CharField(
+        label="Enter 6 digit pin sent via SMS",
+        widget=NumberInput(attrs={"placeholder": "One Time Pin"}),
         required=True,
-        max_length=100,
+        min_length=6,
+        max_length=6,
     )
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request")
+        super().__init__(*args, **kwargs)
+
+    def verify_otp(self, otp):
+        session_otp = self.request.session.get("otp_hash")
+        h = hmac.new(settings.SECRET_KEY.encode(), otp.encode(), digestmod=sha256)
+
+        return hmac.compare_digest(base64.b64encode(h.digest()).decode(), session_otp)
```

Comment:
We should also check that the timestamp (we should store on on the POST to the login view) is within some time limit

Reply:
Yes, will add that as a separate PR.